libraries/Microsoft.Bot.Streaming/Payloads/PayloadStream.cs

```diff
@@ -131,9 +127,9 @@ public override int Read(byte[] buffer, int offset, int count)
 
             if (_active == null)
             {
-                dataAvailable.Wait();
+                _dataAvailable.Wait();
```

Comment:
> _dataAvailable.Wait(); [](start = 15, length = 23)

eeesh, can't we throw an NotSupported exception on this?

Reply:
If the binary compatibility check will let us get away with it, NotSupported would make more sense. I think this was originally added when the class was still internal and we thought it may be useful to have. Now it's just a bear trap sitting in the middle of the hallway. At the least we could add a warning to the summary.